ejb-txn-remote-call/README.adoc

```diff
@@ -0,0 +1,942 @@
+include::../shared-doc/attributes.adoc[]
+
+= ejb-txn-remote-call: Demonstrates remote EJB calls and transaction propagation
+:author: Ondra Chaloupka
+:level: Intermediate
+:technologies: EJB, JTA, Clustering
+
+[abstract]
+The `ejb-txn-remote-call` quickstart demonstrates the remote EJB calls over two application servers of {productName}.
+
+:standalone-server-type: ha
+:archiveType: war
+:uses-h2:
+:requires-multiple-servers:
+:jbds-not-supported:
+
+== What is it?
+
+The `ejb-txn-remote-call` quickstart demonstrates the remote EJB calls over two application servers of {productNameFull}.
+
+== Description
+
+The EJB remote calls contains a JTA transaction which is propagated over the EJB call.
+Further the quickstart demonstrates the transaction recovery which is run for both servers
+when a failure occurs.
+
+This quickstart contains two maven projects. As the goal is to make a remote call from one server
+to the another.
+The first maven projects is meant to be deployed on one server, representing the caller side,
+and the second maven project is meant to be deployed on a set of second servers, representing the callee side.
+
+[cols="40%,60%",options="headers"]
+|===
+|Project |Description
+
+|`client`
+|An application that needs to be deployed to the first server. It contains an EJB which calls
+ the EJB application on the second server.
+ For the EJB processing will be enlisted to the JTA transaction and processed with the two-phase
+ commit there is an execution which inserts data into database.
+
+|`server`
+|An application which is about tobe deployed to the second server. It contains an EJB which is capable
+ to receive a remote call from the first server.
+ For the EJB processing being enlisted to the JTA transaction there is a database insertion
+ being part of the logic in the EJB business method. Then there is a special XAResource
+ enlisted to the transaction for purposes of this quickstart demonstration.
+
+|===
+
+// System Requirements
+include::../shared-doc/system-requirements.adoc[leveloffset=+1]
+// Use of {jbossHomeName}_1 and {jbossHomeName}_2
+include::../shared-doc/use-of-jboss-home-name.adoc[leveloffset=+1]
+// Install the Quickstart Parent Artifact in Maven
+include::../shared-doc/install-quickstart-parent-artifact.adoc[leveloffset=+1]
+
+== Setup the environment
+
+For the EJB remote calls works there is needed to setup all {productName} servers. Let's see
```

Comment:
"You must set up all servers to use the EJB remote calls."

Reply:
+1